core/src/test/kotlin/io/specmatic/core/pattern/EmailPatternTest.kt

```diff
@@ -1,12 +1,18 @@
 package io.specmatic.core.pattern
 
 import io.specmatic.GENERATION
+import io.specmatic.conversions.OpenApiSpecification
+import io.specmatic.core.EXAMPLES_DIR_SUFFIX
+import io.specmatic.core.HttpRequest
 import io.specmatic.core.Resolver
 import io.specmatic.core.Result
+import io.specmatic.mock.ScenarioStub
+import io.specmatic.stub.HttpStub
 import io.specmatic.core.value.*
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.jupiter.api.Tag
 import org.junit.jupiter.api.Test
+import java.io.File
 
 class EmailPatternTest {
     @Test
```

Comment:
Can you also add tests for `matches()` method (success / failure) and `newBasedOn()` method.